src/primitives/torus.js

```diff
@@ -0,0 +1,45 @@
+const {defaultResolution2D, defaultResolution3D} = require('../core/constants')
+
+const {extrudeRotate} = require('../operations')
+
+const circle = require('./circle')
+
+/** Construct a torus by revolving a small circle (inner) about the circumference of a large (outer) circle.
+ * @param {Object} [options] - options for construction
+ * @param {Float} [options.innerRadius=1] - radius of small (inner) circle
+ * @param {Float} [options.outerRadius=4] - radius of large (outer) circle
+ * @param {Integer} [options.innerResolution=defaultResolution2D] - number of segments per 360 rotation
+ * @param {Integer} [options.outerResolution=defaultResolution3D] - number of segments per 360 rotation
+ * @param {Integer} [options.roti=0] - rotation angle of base circle
+ * @returns {geom3} new 3D geometry
+ *
+ * @example
+ * let torus1 = torus({
+ *   innerRadius: 10
+ * })
+ */
+function torus (options) {
+  const defaults = {
+    innerRadius: 1,
+    innerResolution: defaultResolution2D,
+    outerRadius: 4,
+    outerResolution: defaultResolution3D,
+    roti: 0
```

Comment:
hmm you are right, this really is not clear enough

Reply:
Here's a couple of examples from V1.

```
    let obj1 = torus({ fni:4,fno:4,roti:45 });
    let obj2 = torus({ fni:4,fno:4});
```
So, 'roti' is a rotation of the circle, before creating the torus.

Reply:
If we keep this, then possibly name it... 'innerRotation'?